### PR TITLE
fix PPA comp. error and add ubuntu:bionic gcc-7 to CI to test for it.

### PR DESCRIPTION
**Pull request by Marenz**

### CODE REVIEW

**Comment by leonardoalt** [comment]
What's the motivation?

**Comment by Marenz** [comment]
The build currently fails on ubuntu bionic gcc-7 as reported by someone on the chat. Our release for that ubuntu version is broken

**Comment by Marenz** [comment]
Note that this PR is not ready for merging, still a WIP

**Comment by leonardoalt** [comment]
Ah ok

**Comment by Marenz** [comment]
This is now ready

File: .circleci/config.yml [hunk 1]
```diff
@@ -233,6 +233,18 @@ jobs:
       - store_artifacts: *artifacts_solc
       - persist_to_workspace: *artifacts_executables
 
+  b_ubu18: &build_ubuntu1804
+    docker:
+      - image: ethereum/solidity-buildpack-deps:ubuntu1804
+    environment:
```
Comment (leonardoalt) [inline .circleci/config.yml:239]: Why the extra flags? (compared to `b_ubu`)

Comment (Marenz) [reply .circleci/config.yml:239]: the compilation error in that I fixed in this PR as well is only triggered with those flags.

Comment (Marenz) [reply .circleci/config.yml:239]: Those are also the flags we use for the PPA release. At least the relevant ones.

File: libsolidity/codegen/YulUtilFunctions.cpp [hunk 2]
```diff
@@ -922,7 +922,7 @@ string YulUtilFunctions::readFromCalldata(Type const& _type)
 	return readFromMemoryOrCalldata(_type, true);
 }
 
-string YulUtilFunctions::updateStorageValueFunction(Type const& _type, boost::optional<unsigned> const _offset)
+string YulUtilFunctions::updateStorageValueFunction(Type const& _type, boost::optional<unsigned> const& _offset)
```
Comment (bshastry) [inline libsolidity/codegen/YulUtilFunctions.cpp:925]: I'm guessing this change and the one below it does not belong to this PR. Could it be moved to #7077 ?

Comment (Marenz) [reply libsolidity/codegen/YulUtilFunctions.cpp:925]: Without this change, this PR would fail in the CI. Basically, this is the problem that was previously undetected that the CI will now test for

Comment (Marenz) [reply libsolidity/codegen/YulUtilFunctions.cpp:925]: so it does indeed belong together

**Comment by Marenz** [comment]
CI works now

File: libsolidity/codegen/YulUtilFunctions.cpp (hunk 2 above)
Comment (bshastry) [reply libsolidity/codegen/YulUtilFunctions.cpp:925]: okay, looks good to me. may be good to edit PR title to something "Fix build error and add ... gcc-7"

**Review by bshastry — APPROVED** [review]
